test/users.test.js

```diff
@@ -0,0 +1,93 @@
+const request = require('supertest');
+const truncateDatabase = require('./setup');
+const app = require('../app');
+
+beforeAll(async () => {
+  process.env.NODE_ENV = 'test';
+  await truncateDatabase();
+});
+
+describe('POST users', () => {
+  const newUserData = {
+    name: 'William',
+    lastName: 'Salazar',
+    email: 'william.salazar@wolox.co',
+    password: 'thisIsAVerySecurePassword'
+  };
+
+  test('create new user', async done => {
+    const response = await request(app)
+      .post('/users')
+      .set('Accept', 'application/json')
+      .send(newUserData);
+
+    const {
+      status,
+      body: { name }
+    } = response;
+
+    expect(status).toBe(201);
+    expect(name).toBe('William');
+    done();
+  });
+
+  test('Should return an error when email is already in use', async done => {
+    const response = await request(app)
+      .post('/users')
+      .set('Accept', 'application/json')
+      .send(newUserData);
+
+    const {
+      status,
+      body: { message }
+    } = response;
+
+    expect(status).toBe(400);
+    expect(message).toBe('Email address already in use!');
```

Comment:
You should create a constant file, that contains all error messages. And then import it as error messages in the model and the tests.

Reply:
Done